Tests/ClientRuntimeTests/NetworkingTests/HashFunctionTests.swift

```diff
@@ -0,0 +1,163 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import XCTest
+@testable import ClientRuntime
+import AwsCommonRuntimeKit
+
+class HashFunctionTests: XCTestCase {
+
+    override func setUp() {
+        // Initialize function needs to be called before interacting with CRT
+        CommonRuntimeKit.initialize()
+    }
+
+    func testCRC32NonUTF8Bytes() {
+        guard let hashFunction = HashFunction.from(string: "crc32") else {
+            XCTFail("CRC32 not found")
+            return
+        }
```

Comment:
Debatable: Feel free to ignore. 
I find it better and more concise to use `let hashFunction = HashFunction.from(string: "crc32")!` instead of a guard in tests. The test will fail anyway if we don't find it. Same comment for all the guards, `.?`, and `try?` in these tests.

Reply:
removing `try?` causes error unhandled errors, the guard statements I prefer to keep in for precise error messaging should it ever fail

Reply:
Okay, by the way, you can replace `try?` with `try!` and `.?` with `.!` to avoid causing any errors.

Reply:
Generally agreed, I am less of a stickler about force-unwrap in tests, though it can be annoying for the test suite to crash when it could have just failed a test & moved on.

If the `!` is expected to fail under any circumstance, I'd prefer to throw & fail versus crash the test suite.

Reply:
Good to know! For now I am going to continue with the guard so that it is abundantly clear what the issue is if we get a failure here